src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -61,15 +63,34 @@ export const canShow = async (
         );
 
         appboy.initialize(apiKey, {
+            enableLogging: false,
             noCookies: true,
             baseUrl: 'https://sdk.fra-01.braze.eu/api/v3',
             sessionTimeoutInSeconds: 1,
             minimumIntervalBetweenTriggerActionsInSeconds: 0,
         });
 
         return new Promise((resolve) => {
-            appboy.subscribeToInAppMessage((message) => {
-                const meta = (message as any).extras;
+            appboy.subscribeToInAppMessage((message: any) => {
+                const meta = message.extras;
+
+                buttonHandler = (buttonId: number) => {
+                    const thisButton = new appboy.InAppMessageButton(
+                        `Button: ID ${buttonId}`,
+                        undefined,
+                        undefined,
+                        undefined,
+                        undefined,
+                        undefined,
+                        buttonId,
+                    );
+                    appboy.logInAppMessageButtonClick(thisButton, message);
+                };
+
+                if (appboy) {
+                    // Log the impression with Braze
+                    appboy.logInAppMessageImpression(message);
+                }
```

Comment:
I think the right way to expose the click handler and the impression log functionality to the component is via the `meta` object which is threaded through to the component as a prop. So `meta` could be extended to included two additional fields:

```js
{
  logImpression: () => { appboy.logInAppMessageImpression(message) },
  buttonHandler: (buttonId: number) => {...},
}
```

In general having React components rely on dependencies which aren't passed as props (which is what we're doing with `buttonHandler` here) can cause confusing situations as they exist outside of React's lifecycle. It's definitely confusing though because this is how we thread things together on frontend (where we're not in React world).

Really happy to pair on this if it'd be useful!

Reply:
Cheers Tom! I've pushed the fix we made for logImpression earlier today.

Reply:
Also have added the `buttonHandler` to the meta prop.